api/pkg/pubsub/nats.go

```diff
@@ -22,7 +23,13 @@ type Nats struct {
 	consumer   jetstream.Consumer
 }
 
-func NewInMemoryNats(storeDir string) (*Nats, error) {
+func NewInMemoryNats() (*Nats, error) {
+
+	// disregard storeDir, it causes "filtered consumer not unique on workqueue
+	// stream" error on restarts. For now, just use a random temp directory so
+	// we are using the disk, but not the same directory on restarts.
+	storeDir := os.TempDir()
```

Comment:
Keep in mind, once you do this, you "own" this directory, meaning you need to make sure you clean it up once done with it...

The OG code passes `t.TempDir()` which is automatically cleaned up by Go's testing machinery: https://pkg.go.dev/testing#T.TempDir

Reply:
Another reason, this might _not_ work is, if you dont clean up that `os.TmpDir` you might keep trying to use the same dir across all tests meaning the JetStream data wont be unique per tests and _things happen_